lib/src/switch/Switch.tsx

```diff
@@ -228,12 +169,134 @@ const LabelContainer = styled.span`
   font-style: ${(props) => (props.disabled ? props.theme.disabledLabelFontStyle : props.theme.labelFontStyle)};
   font-weight: ${(props) => props.theme.labelFontWeight};
   cursor: ${(props) => (props.disabled === true ? "not-allowed" : "pointer")};
+
   ${(props) =>
-    props.labelPosition === "after"
+    !props.hasLabel
+      ? "margin: 0px;"
+      : props.labelPosition === "after"
       ? `margin-left: ${props.theme.spaceBetweenLabelSwitch};`
-      : `margin-right: ${props.theme.spaceBetweenLabelSwitch};`}
+      : `margin-right: ${props.theme.spaceBetweenLabelSwitch};`};
 
   ${(props) => props.labelPosition === "before" && "order: -1"}
 `;
 
+type SwitchBaseProps = {
+  labelPosition: "after" | "before";
+  hasLabel: boolean;
+};
+
+const SwitchBase = styled.label<SwitchBaseProps>`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  cursor: pointer;
+  margin: ${(props) =>
+    !props.hasLabel ? "0px 4px" : props.labelPosition === "before" ? "0 4px 0 12px" : "0 12px 0 4px"};
+`;
+
+const SwitchInput = styled.input`
+  opacity: 0;
+  width: 0;
+  height: 0;
+  margin: 0px;
+`;
+
+type SwitchTrackProps = {
+  backgroundType: "dark" | "light";
+};
+
+const SwitchTrack = styled.span<SwitchTrackProps>`
+  background-color: red;
```

Comment:
What is this color red for?

Reply:
I used it for testing purposes. Going to be removed.